driver-sync/src/test/functional/com/mongodb/client/model/mql/AggregatesSearchTest.java

```diff
@@ -0,0 +1,249 @@
+                                + "   { _id: 8, summary: 'summary 1' },\n"
+                                + "   { _id: 9, summary: 'summary 2' },\n"
+                                + "]");
+    }
+
+    @Test
+    public void testRegex() {
+        List<Bson> pipeline = Arrays.asList(
+                search(SearchOperator.regex(fieldPath("description"), "des[c]+ <1-4>"),
+                        searchOptions().index(searchIndexName)));
+        assertResults(pipeline, "[\n"
+                                + "   { _id: 6, description: 'desc 1' },\n"
+                                + "]");
+    }
+
+    @Test
+    public void testWildcard() {
+        List<Bson> pipeline = Arrays.asList(
+                search(SearchOperator.wildcard(fieldPath("description"), "desc*"),
+                        searchOptions().index(searchIndexName)));
+        assertResults(pipeline, "[\n"
+                                + "   { _id: 6, description: 'desc 1' },\n"
+                                + "   { _id: 7, description: 'desc 8' },\n"
+                                + "]");
+    }
+
+    private static void insertDocuments(final String s) {
+        List<Document> documents = BsonArray.parse(s).stream()
+                .map(v -> new Document(v.asDocument()))
+                .collect(Collectors.toList());
+        collection.insertMany(documents);
+    }
+
+    private static void assertResults(final List<Bson> pipeline, final String expectedResultsAsString) {
+        ArrayList<Bson> pipeline2 = new ArrayList<>(pipeline);
+        pipeline2.add(sort(ascending("_id")));
+
+        List<BsonDocument> expectedResults = parseToList(expectedResultsAsString);
+        List<BsonDocument> actualResults = aggregate(pipeline2);
+        assertEquals(expectedResults, actualResults);
+    }
+
+    private static List<BsonDocument> aggregate(final List<Bson> stages) {
+        AggregateIterable<Document> result = collection.aggregate(stages);
+        List<BsonDocument> results = new ArrayList<>();
+        result.forEach(r -> results.add(r.toBsonDocument()));
+        return results;
+    }
+
+    public static List<BsonDocument> parseToList(final String s) {
+        return BsonArray.parse(s).stream().map(v -> toBsonDocument(v.asDocument())).collect(Collectors.toList());
+    }
+
+    public static BsonDocument toBsonDocument(final BsonDocument bsonDocument) {
+        return getDefaultCodecRegistry().get(BsonDocument.class).decode(bsonDocument.asBsonReader(), DecoderContext.builder().build());
+    }
+
+    @Nullable
+    private static Document indexRecord(
+            final MongoCollection<Document> collection, final String indexName) {
+        return StreamSupport.stream(collection.listSearchIndexes().spliterator(), false)
+                .filter(index -> indexName.equals(index.getString("name")))
+                .findAny().orElse(null);
+    }
+
+    static void waitForIndex(final MongoCollection<Document> collection, final String indexName) {
```

Comment:
This does duplicate functionality from the other test - it copies what I used in LangChain4j, and handles FAILED. I did not merge both, because that would require moving the other test to `sync`. I do not mind doing this, if that seems best.